internal/exec/engine.go

```diff
@@ -70,12 +72,11 @@ func (e Engine) Run(stageName string) error {
 	systemBaseConfig, r, err := system.FetchBaseConfig(e.Logger)
 	e.logReport(r)
 	if err != nil && err != providers.ErrNoProvider {
-		e.Logger.Crit("failed to acquire system base config: %v", err)
-		return err
```

Comment:
Hmm, not sure if there's a reason why we would both log a message and return an error. That pattern is used elsewhere as well. @bgilbert Do you know offhand?

Reply:
I don't _know_, but I suspect it's because the returned error doesn't currently include any context on what the caller was doing.